shard/connpool.lua

```diff
@@ -112,60 +166,31 @@ local function zone_list(self)
     return names
 end
 
-local function _on_disconnect(self, srv)
-    self:on_disconnect_one(srv)
-
-    -- check zone and pool
-    local d = 0
-    local all_zones = self:zone_list()
-    for _, name in pairs(all_zones) do
-        local alive = self:all(name)
-        if #alive == 0 then
-            self:on_disconnect_zone(name)
-            d = d + 1
-        end
-    end
-    if d == #all_zones then
-        self:on_disconnect()
-    end
-end
-
 local function monitor_fiber(self)
-    fiber.name("monitor")
+    fiber.name("_" .. self.configuration.pool_name .. "_monitor")
     local i = 0
     while true do
         i = i + 1
-        local server = self:one(nil, true)
+        local server = self:get_any_active_server(nil, true)
 
-        if server ~= nil then
+        if server.state == 'connected' then
```

Comment:
Again, you checks state directly, but you have server_is_ok function. Moreover, I think, that you can add a metatable to a server object, in which server_is_ok can be added together with server_is_disconnected, server_is_dead etc, and you will call them just like
```Lua
server:is_ok()
server:is_connected()
server:is_dead()
-- ...
```